File: c++/17073.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> adjList[500001];
bool isVisited[500001];

double cnt;

void dfs(int n) {
  int cnCnt = 0;
  for (int i = 0; i < adjList[n].size(); i++) {
    int nn = adjList[n][i];

    if (!isVisited[nn]) {
      cnCnt++;
      isVisited[nn] = true;
      dfs(nn);
      isVisited[nn] = false;
    }
  }

  if (cnCnt == 0) cnt++;
}

int main() {
  freopen("example.txt", "r", stdin);

  int N, W;
  cin >> N >> W;

  int a, b;
  while (cin >> a >> b) {
    adjList[a].push_back(b);
    adjList[b].push_back(a);
  }

  isVisited[1] = true;
  dfs(1);

  cout.precision(10);

  cout << W / cnt;

  return 0;
}

File: js/17073.js
const fs = require('fs');

const inputs = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split('\n')
  .map((e) => e.split(' ').map(Number));

const [N, W] = inputs[0];
let cnt = 0;

const adjList = Array(N + 1)
  .fill()
  .map(() => []);
const isVisited = Array(N + 1).fill(false);

for (let i = 1; i < inputs.length; i++) {
  const [a, b] = inputs[i];
  adjList[a].push(b);
  adjList[b].push(a);
}

isVisited[1] = true;
dfs(1);
console.log(W / cnt);

function dfs(n) {
  let cnCnt = 0;
  for (let i = 0; i < adjList[n].length; i++) {
    const nn = adjList[n][i];
    if (!isVisited[nn]) {
      cnCnt++;
      isVisited[nn] = true;
      dfs(nn);
      isVisited[nn] = false;
    }
  }
  if (cnCnt === 0) cnt++;
}
